openedx/features/course_experience/views/course_outline.py

```diff
@@ -168,11 +170,22 @@ def reset_course_deadlines(request, course_id):
     Set the start_date of a schedule to today, which in turn will adjust due dates for
     sequentials belonging to a self paced course
     """
-    course = CourseOverview.objects.get(id=course_id)
+    course_key = CourseKey.from_string(course_id)
+    course = CourseOverview.objects.get(id=course_key)
     if course.self_paced:
-        enrollment = CourseEnrollment.objects.get(user=request.user, course=course_id)
+        masquerade_details, masquerade_user = setup_masquerade(
+            request,
+            course_key,
+            has_access(request.user, 'staff', course, course_key)
+        )
+        if masquerade_details and masquerade_details.role == 'student' and masquerade_details.user_name:
+            # Masquerading as a specific student, so reset that student's schedule
+            user = masquerade_user
+        else:
+            user = request.user
+        enrollment = CourseEnrollment.objects.get(user=user, course=course_key)
         schedule = enrollment.schedule
         if schedule:
             schedule.start_date = datetime.datetime.now(pytz.utc)
             schedule.save()
-    return redirect(reverse('openedx.course_experience.course_home', args=[six.text_type(course_id)]))
+    return redirect(reverse('openedx.course_experience.course_home', args=[six.text_type(course_key)]))
```

Comment:
whats the difference between using course_key, and course_id here?

Reply:
Conventionally, `course_key` is the `CourseKey` object, and `course_id` is the string representation of that key.